### DISCUSSION TITLE
Filtering What's Happening content

### DISCUSSION THREAD

**Discussion by sfederici** [ScratchAddons/ScratchAddons#1593] [Ideas]
I love the possibility to see more "What's Happening" content. But I'm not interested at all to "x is following y" notifications and some F4F enthusiasts clog the content of the area with tens of "x is follwing y" notifications. It would be great if one could filter out the events they are not interested in.

**Comment by sfederici** [comment]
NOTE: updated to **version 2.3.9**

I created a [scratchaddon](https://github.com/user-attachments/files/30548981/full-areas-whats-happening-integrated-v2.3.9.zip) using chatgpt to perform this task. It is a modification of a  [chrome extension](https://github.com/user-attachments/files/30538688/scratch-whats-happening-filter-v2.3.9.zip) I created with chatgpt to add this possibility to chrome, that is:

- lines can be filtered out for type of event (liked, favorited, shared/remixed, olthers)
- lines are shown only if the event is not related to projects alredy visited/liked/favorited ("new" checkbox) 
- it shows slightly more lines than the standard "full areas" addon (10 more lines every time you click the "load more" button)
- filter settings are stored in the browser

<img width="611" height="406" alt="image" src="https://github.com/user-attachments/assets/7576dc2b-8aaf-49af-ae49-11b359754112" />

NOTE: the chrome extension is compatible with the "full areas" addon

The extension (english only) works perfectly. I don't know about the addon (it is supposed to work well with the internal translation of Scratch/addons)

**Comment by Samq64** [comment]
Someone implemented this in #4701 but it was abandoned.

**Reply by sfederici** [reply]
It seems to me that it was exactly what I needed. Why was it abandoned? Problems of compatibility with ScratchAddons?

**Comment by sfederici** [comment]
BTW, my addon, if it works, has in addition removal of lines regarding love/favorite of a same project by more Scratchers and removal (if "new" is checked) of projects already visited/loved/favorited by the user. The "visited" feature is based on the browser history.
